classes/Context.php

```diff
@@ -332,12 +332,13 @@ public function updateCustomer(Customer $customer)
         $customer->logged = 1;
         $this->cookie->email = $customer->email;
         $this->cookie->is_guest = $customer->isGuest();
-        $this->cart->secure_key = $customer->secure_key;
 
         if (Configuration::get('PS_CART_FOLLOWING') && (empty($this->cookie->id_cart) || Cart::getNbProducts($this->cookie->id_cart) == 0) && $idCart = (int) Cart::lastNoneOrderedCart($this->customer->id)) {
             $this->cart = new Cart($idCart);
+            $this->cart->secure_key = $customer->secure_key;
         } else {
```

Comment:
Would it make sense to add another validation layer by adding something like `if ($this->cart !== null)` ?

Reply:
I don't think so, new Cart() should always return a cart, or throw an Exception

Reply:
I was targeting the `else` case 😄 but the github UI makes it look like I show the `if`

Reply:
Why not adding this on line 348 after 
https://github.com/PrestaShop/PrestaShop/pull/14352/files#diff-c606149f06205f3d2db241862eb4e84eR348?

Reply:
I tried that before but Travis was failing so I try with this approach, perhaps the upcoming function calls need the secure key